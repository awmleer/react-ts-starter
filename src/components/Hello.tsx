import * as React from "react"
import {Button} from 'antd'

interface Props { compiler: string; framework: string; }

// 'Props' describes the shape of props.
// State is never set so we use the '{}' type.
export class Hello extends React.Component<Props, {}> {
  x(y: any) {
    console.log(y)
  }

  render() {
    console.log(this.x`123`)
    return (
      <div>
        <h1>Hello from {this.props.compiler} and {this.props.framework}!</h1>
        <Button type="primary">Test Button 2</Button>
      </div>
    )
  }
}
