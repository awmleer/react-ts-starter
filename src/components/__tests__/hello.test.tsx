import * as React from 'react'
import * as renderer from 'react-test-renderer'
import {Hello} from '../Hello'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Hello compiler="TypeScript 123" framework="React" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
