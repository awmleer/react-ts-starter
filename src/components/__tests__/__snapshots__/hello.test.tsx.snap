// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Hello from 
    TypeScript 123
     and 
    React
    !
  </h1>
  <button
    className="ant-btn ant-btn-primary"
    onClick={[Function]}
    type="button"
  >
    <span>
      Test Button 2
    </span>
  </button>
</div>
`;
